File: java/chaincue-real-estate/src/main/java/com/chaincuerealestate/chaincuerealestate/routes/house/HousePageDTO.java
package com.chaincuerealestate.chaincuerealestate.routes.house;

public record HousePageDTO(
        String id,
        String title,
        String type,
        String location,
        int numberOfRooms,
        int beds,
        String dollarPrice,
        String cryptoPrice,
        String description,
        HouseImage[] images,
        Broker broker
) {

    public record HouseImage(
            String id,
            String url
    ) {
    }

    public record Broker(
            String id,
            String name,
            String phoneNumber,
            String email
    ) {
    }

}

File: kotlin/chaincue-real-estate/src/main/kotlin/com/example/chaincuerealestate/routes/house/HousePageDTO.kt
package com.example.chaincuerealestate.routes.house

data class HousePageDTO(
    val id: String,
    val title: String,
    val type: String,
    val location: String,
    val numberOfRooms: Int,
    val beds: Int,
    val dollarPrice: String,
    val cryptoPrice: String,
    val description: String,
    val images: Array<HouseImage>,
    val broker: Broker?
) {
    data class HouseImage(
        val id: String,
        val url: String
    )

    data class Broker(
        val id: String,
        val name: String,
        val phoneNumber: String,
        val email: String
    )
}
